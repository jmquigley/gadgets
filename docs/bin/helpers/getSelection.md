<a name="getSelectionShim"></a>

## getSelectionShim()
DOM Traversal is not implemented in JSDOM
The best we can do is shim the functions

**Kind**: global function  
