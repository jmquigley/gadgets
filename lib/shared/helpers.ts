"use strict";

const debug = require("debug")("helpers");

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is already in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(
	name: string,
	pkgname: any,
	replace: boolean = false
) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
		debug("Adding global variable %o -> %O", name, (global as any)[name]);
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
