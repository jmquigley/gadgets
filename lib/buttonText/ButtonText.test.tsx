'use strict';

import {cleanup, log, mockupEnv} from '../../test/helpers';
mockupEnv();

import test from 'ava';
import {mount} from 'enzyme';
import * as path from 'path';
import * as React from 'react';
import * as sinon from 'sinon';
import {ButtonText, getDefaultButtonTextProps} from './index';

test.after.always.cb(t => {
	cleanup(path.basename(__filename), t);
});

test('Test retrieval of ButtonText props object', t => {
	const props = getDefaultButtonTextProps();

	t.true('iconName' in props);
	t.is(props.iconName, 'bomb');

	t.true('justify' in props);
	t.is(props.justify, ButtonText.RIGHT);

	t.true('text' in props);
	t.is(props.text, '');
});

test('Test creation of a ButtonText control', t => {
	const ctl = mount(
		<ButtonText
		  className="test-class"
		  text="test text"
		  justify={ButtonText.LEFT}
		  color="white"
		  borderColor="green"
		  backgroundColor="blue"
		  />
	);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.is(ctl.prop('id'), '');
	t.false(ctl.prop('disabled'));
	t.true(ctl.prop('visible'));
	t.is(ctl.prop('text'), "test text");
	t.is(ctl.prop('justify'), ButtonText.LEFT);
	t.is(ctl.prop('color'), "white");
	t.is(ctl.prop('backgroundColor'), "blue");
	t.is(ctl.prop('borderColor'), "green");

	t.is(ctl.find('.test-class').length, 1);
	t.is(ctl.find('.ui-button').length, 1);
	t.is(ctl.find('.buttonText').length, 1);
});

test('Test button click in ButtonText control', t => {
	let click = sinon.spy();
	const ctl = mount(
		<ButtonText
		  className="test-class"
		  text="test text"
		  onClick={click}
		  />
	);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.false(ctl.prop('disabled'));
	t.true(ctl.prop('visible'));
	t.is(ctl.prop('text'), "test text");

	let btn = ctl.find('.ui-buttontext');
	t.is(btn.length, 1);
	btn.simulate('click');
	t.true(click.calledOnce);
});

test('Test disabling of a ButtonText control', t => {
	const click = sinon.spy();
	const ctl = mount(<ButtonText onClick={click} disabled={true} />);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.is(ctl.prop('iconName'), 'bomb');
	t.true(ctl.prop('disabled'));
	t.true(ctl.prop('visible'));
	t.is(ctl.find('.disabled').length, 1);

	ctl.find('.ui-buttontext').simulate('click');
	t.true(click.neverCalledWith());
});

test('Test making a ButtonText control invisible', t => {
	const click = sinon.spy();
	const ctl = mount(<ButtonText onClick={click} visible={false} />);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.is(ctl.prop('iconName'), 'bomb');
	t.false(ctl.prop('disabled'));
	t.false(ctl.prop('visible'));
	t.is(ctl.find('.invisible').length, 1);

	ctl.find('.ui-buttontext').simulate('click');
	t.true(click.neverCalledWith());
});
