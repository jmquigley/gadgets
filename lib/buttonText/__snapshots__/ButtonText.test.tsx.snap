// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a ButtonText control to the left 1`] = `
<ButtonText
  bottom=""
  className="test-class"
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage=""
  focus={false}
  height=""
  iconName="bomb"
  id=""
  imageFile=""
  justify={0}
  left=""
  location="none"
  noedit={false}
  nohover={false}
  noicon={false}
  noripple={false}
  obj="ButtonText"
  onClick={[Function]}
  padding=""
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "blue",
      "borderColor": "green",
      "color": "white",
    }
  }
  testing={true}
  text="test text"
  theme={null}
  tooltip=""
  top=""
  visible={true}
  width=""
  xcss={null}
>
  <Wrapper
    bottom=""
    className="test-class"
    contentEditable={false}
    controlled={true}
    direction="right"
    disabled={false}
    err={null}
    errorMessage=""
    focus={false}
    height=""
    iconName="bomb"
    id=""
    imageFile=""
    justify={0}
    left=""
    location="none"
    noedit={false}
    nohover={false}
    noicon={false}
    noripple={false}
    obj="ButtonText"
    onClick={[Function]}
    onError={[Function]}
    padding=""
    reset={false}
    selected={false}
    sizing="normal"
    style={
      Object {
        "backgroundColor": "blue",
        "borderColor": "green",
        "color": "white",
      }
    }
    testing={true}
    text="test text"
    theme={null}
    tooltip=""
    top=""
    visible={true}
    width=""
    xcss={null}
  >
    <ThemeProvider
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#c0c0c0",
          "color": "#000000",
          "headerBackgroundColor": "#004358",
          "headerForegroundColor": "#ffffff",
          "headerHoverColor": "#ffe11a",
          "hoverColor": "#c0c0c0",
          "inputBorderColor": "#c0c0c0",
          "itemHoverColor": "#bedb39",
          "selectedBackgroundColor": "#1f8a70",
          "selectedForegroundColor": "#ffe11a",
          "titleBarBackgroundColor": "#004358",
          "titleBarForegroundColor": "#ffe11a",
          "tooltipBackgroundColor": "#708090",
          "tooltipForegroundColor": "#ffffff",
          "transitionDelay": "0.5s",
        }
      }
    >
      <styled.div
        className="ui-button-text ripple test-class"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "blue",
            "borderColor": "green",
            "color": "white",
          }
        }
        visible={true}
      >
        <StyledComponent
          className="ui-button-text ripple test-class"
          disabled={false}
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-kgoBCf",
                "isStatic": false,
                "lastClassName": "dOBixR",
                "rules": Array [
                  "
	",
                  "
	align-items: center;
	box-sizing: border-box;
	cursor: default;
	display: flex;
	height: 100%;
	justify-content: center;
	line-height: inherit;
	outline: none;
	overflow: visible;
	position: relative;
	user-select: none;
	width: 100%;
",
                  "

	&:not(.nohover):hover {
		background-color: ",
                  [Function],
                  " ",
                  [Function],
                  ";
	}

	",
                  [Function],
                  "
	",
                  [Function],
                  "
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-kgoBCf",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "borderColor": "green",
              "color": "white",
            }
          }
          visible={true}
        >
          <div
            className="ui-button-text ripple test-class sc-kgoBCf dOBixR"
            disabled={false}
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "blue",
                "borderColor": "green",
                "color": "white",
              }
            }
          >
            <styled.div
              justify={0}
              sizing="normal"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-chPdSV",
                      "isStatic": false,
                      "lastClassName": "emANqb",
                      "rules": Array [
                        "
	flex: 1;
	padding: 0 0.2em;
	text-align: ",
                        [Function],
                        ";
	",
                        [Function],
                        ";

	span {
		margin: 0 auto;
		user-select: none;
	}
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-chPdSV",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                justify={0}
                sizing="normal"
              >
                <div
                  className="sc-chPdSV emANqb"
                >
                  test text
                </div>
              </StyledComponent>
            </styled.div>
            <Icon
              bottom=""
              className=""
              contentEditable={false}
              controlled={true}
              direction="right"
              disabled={false}
              err={null}
              errorMessage=""
              focus={false}
              height=""
              iconName="bomb"
              id=""
              imageFile=""
              left=""
              location="none"
              noedit={false}
              nohover={false}
              noripple={false}
              obj="Icon"
              padding=""
              selected={false}
              sizing="normal"
              style={Object {}}
              testing={true}
              theme={null}
              tooltip=""
              top=""
              visible={true}
              width=""
              xcss={null}
            >
              <Wrapper
                bottom=""
                className=""
                contentEditable={false}
                controlled={true}
                direction="right"
                disabled={false}
                err={null}
                errorMessage=""
                focus={false}
                height=""
                iconName="bomb"
                id=""
                imageFile=""
                left=""
                location="none"
                noedit={false}
                nohover={false}
                noripple={false}
                obj="Icon"
                onError={[Function]}
                padding=""
                reset={false}
                selected={false}
                sizing="normal"
                style={Object {}}
                testing={true}
                theme={null}
                tooltip=""
                top=""
                visible={true}
                width=""
                xcss={null}
              >
                <ThemeProvider
                  theme={
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderColor": "#c0c0c0",
                      "color": "#000000",
                      "headerBackgroundColor": "#004358",
                      "headerForegroundColor": "#ffffff",
                      "headerHoverColor": "#ffe11a",
                      "hoverColor": "#c0c0c0",
                      "inputBorderColor": "#c0c0c0",
                      "itemHoverColor": "#bedb39",
                      "selectedBackgroundColor": "#1f8a70",
                      "selectedForegroundColor": "#ffe11a",
                      "titleBarBackgroundColor": "#004358",
                      "titleBarForegroundColor": "#ffe11a",
                      "tooltipBackgroundColor": "#708090",
                      "tooltipForegroundColor": "#ffffff",
                      "transitionDelay": "0.5s",
                    }
                  }
                >
                  <styled.i
                    bottom=""
                    className="ui-icon fa fa-fw fa-bomb"
                    contentEditable={false}
                    controlled={true}
                    direction="right"
                    disabled={false}
                    err={null}
                    errorMessage=""
                    focus={false}
                    height=""
                    iconName="bomb"
                    id=""
                    imageFile=""
                    left=""
                    location="none"
                    noedit={false}
                    nohover={false}
                    noripple={false}
                    obj="Icon"
                    padding=""
                    selected={false}
                    sizing="normal"
                    style={Object {}}
                    testing={true}
                    theme={null}
                    tooltip=""
                    top=""
                    visible={true}
                    width=""
                    xcss={null}
                  >
                    <StyledComponent
                      bottom=""
                      className="ui-icon fa fa-fw fa-bomb"
                      contentEditable={false}
                      controlled={true}
                      direction="right"
                      disabled={false}
                      err={null}
                      errorMessage=""
                      focus={false}
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-gqjmRU",
                            "isStatic": false,
                            "lastClassName": "iCGQxQ",
                            "rules": Array [
                              "
	text-align: center;
	",
                              [Function],
                              "
	",
                              [Function],
                              "
	",
                              [Function],
                              "
	",
                              [Function],
                              "
",
                            ],
                          },
                          "displayName": "styled.i",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-gqjmRU",
                          "target": "i",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      height=""
                      iconName="bomb"
                      id=""
                      imageFile=""
                      left=""
                      location="none"
                      noedit={false}
                      nohover={false}
                      noripple={false}
                      obj="Icon"
                      padding=""
                      selected={false}
                      sizing="normal"
                      style={Object {}}
                      testing={true}
                      theme={null}
                      tooltip=""
                      top=""
                      visible={true}
                      width=""
                      xcss={null}
                    >
                      <i
                        className="ui-icon fa fa-fw fa-bomb sc-gqjmRU iCGQxQ"
                        contentEditable={false}
                        direction="right"
                        disabled={false}
                        height=""
                        id=""
                        selected={false}
                        style={Object {}}
                        width=""
                      />
                    </StyledComponent>
                  </styled.i>
                </ThemeProvider>
              </Wrapper>
            </Icon>
          </div>
        </StyledComponent>
      </styled.div>
    </ThemeProvider>
  </Wrapper>
</ButtonText>
`;

exports[`Test disabling of a ButtonText control 1`] = `
<ButtonText
  bottom=""
  className=""
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={true}
  err={null}
  errorMessage=""
  focus={false}
  height=""
  iconName="bomb"
  id=""
  imageFile=""
  justify={1}
  left=""
  location="none"
  noedit={false}
  nohover={false}
  noicon={false}
  noripple={false}
  obj="ButtonText"
  onClick={[MockFunction]}
  padding=""
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  text=""
  theme={null}
  tooltip=""
  top=""
  visible={true}
  width=""
  xcss={null}
>
  <Wrapper
    bottom=""
    className=""
    contentEditable={false}
    controlled={true}
    direction="right"
    disabled={true}
    err={null}
    errorMessage=""
    focus={false}
    height=""
    iconName="bomb"
    id=""
    imageFile=""
    justify={1}
    left=""
    location="none"
    noedit={false}
    nohover={false}
    noicon={false}
    noripple={false}
    obj="ButtonText"
    onClick={[MockFunction]}
    onError={[Function]}
    padding=""
    reset={false}
    selected={false}
    sizing="normal"
    style={Object {}}
    testing={true}
    text=""
    theme={null}
    tooltip=""
    top=""
    visible={true}
    width=""
    xcss={null}
  >
    <ThemeProvider
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#c0c0c0",
          "color": "#000000",
          "headerBackgroundColor": "#004358",
          "headerForegroundColor": "#ffffff",
          "headerHoverColor": "#ffe11a",
          "hoverColor": "#c0c0c0",
          "inputBorderColor": "#c0c0c0",
          "itemHoverColor": "#bedb39",
          "selectedBackgroundColor": "#1f8a70",
          "selectedForegroundColor": "#ffe11a",
          "titleBarBackgroundColor": "#004358",
          "titleBarForegroundColor": "#ffe11a",
          "tooltipBackgroundColor": "#708090",
          "tooltipForegroundColor": "#ffffff",
          "transitionDelay": "0.5s",
        }
      }
    >
      <styled.div
        className="ui-button-text nohover"
        disabled={true}
        onClick={[Function]}
        style={Object {}}
        visible={true}
      >
        <StyledComponent
          className="ui-button-text nohover"
          disabled={true}
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-kgoBCf",
                "isStatic": false,
                "lastClassName": "dkAsaB",
                "rules": Array [
                  "
	",
                  "
	align-items: center;
	box-sizing: border-box;
	cursor: default;
	display: flex;
	height: 100%;
	justify-content: center;
	line-height: inherit;
	outline: none;
	overflow: visible;
	position: relative;
	user-select: none;
	width: 100%;
",
                  "

	&:not(.nohover):hover {
		background-color: ",
                  [Function],
                  " ",
                  [Function],
                  ";
	}

	",
                  [Function],
                  "
	",
                  [Function],
                  "
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-kgoBCf",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          onClick={[Function]}
          style={Object {}}
          visible={true}
        >
          <div
            className="ui-button-text nohover sc-kgoBCf dkAsaB"
            disabled={true}
            onClick={[Function]}
            style={Object {}}
          >
            <Icon
              bottom=""
              className=""
              contentEditable={false}
              controlled={true}
              direction="right"
              disabled={false}
              err={null}
              errorMessage=""
              focus={false}
              height=""
              iconName="bomb"
              id=""
              imageFile=""
              left=""
              location="none"
              noedit={false}
              nohover={false}
              noripple={false}
              obj="Icon"
              padding=""
              selected={false}
              sizing="normal"
              style={Object {}}
              testing={true}
              theme={null}
              tooltip=""
              top=""
              visible={true}
              width=""
              xcss={null}
            >
              <Wrapper
                bottom=""
                className=""
                contentEditable={false}
                controlled={true}
                direction="right"
                disabled={false}
                err={null}
                errorMessage=""
                focus={false}
                height=""
                iconName="bomb"
                id=""
                imageFile=""
                left=""
                location="none"
                noedit={false}
                nohover={false}
                noripple={false}
                obj="Icon"
                onError={[Function]}
                padding=""
                reset={false}
                selected={false}
                sizing="normal"
                style={Object {}}
                testing={true}
                theme={null}
                tooltip=""
                top=""
                visible={true}
                width=""
                xcss={null}
              >
                <ThemeProvider
                  theme={
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderColor": "#c0c0c0",
                      "color": "#000000",
                      "headerBackgroundColor": "#004358",
                      "headerForegroundColor": "#ffffff",
                      "headerHoverColor": "#ffe11a",
                      "hoverColor": "#c0c0c0",
                      "inputBorderColor": "#c0c0c0",
                      "itemHoverColor": "#bedb39",
                      "selectedBackgroundColor": "#1f8a70",
                      "selectedForegroundColor": "#ffe11a",
                      "titleBarBackgroundColor": "#004358",
                      "titleBarForegroundColor": "#ffe11a",
                      "tooltipBackgroundColor": "#708090",
                      "tooltipForegroundColor": "#ffffff",
                      "transitionDelay": "0.5s",
                    }
                  }
                >
                  <styled.i
                    bottom=""
                    className="ui-icon fa fa-fw fa-bomb"
                    contentEditable={false}
                    controlled={true}
                    direction="right"
                    disabled={false}
                    err={null}
                    errorMessage=""
                    focus={false}
                    height=""
                    iconName="bomb"
                    id=""
                    imageFile=""
                    left=""
                    location="none"
                    noedit={false}
                    nohover={false}
                    noripple={false}
                    obj="Icon"
                    padding=""
                    selected={false}
                    sizing="normal"
                    style={Object {}}
                    testing={true}
                    theme={null}
                    tooltip=""
                    top=""
                    visible={true}
                    width=""
                    xcss={null}
                  >
                    <StyledComponent
                      bottom=""
                      className="ui-icon fa fa-fw fa-bomb"
                      contentEditable={false}
                      controlled={true}
                      direction="right"
                      disabled={false}
                      err={null}
                      errorMessage=""
                      focus={false}
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-gqjmRU",
                            "isStatic": false,
                            "lastClassName": "iCGQxQ",
                            "rules": Array [
                              "
	text-align: center;
	",
                              [Function],
                              "
	",
                              [Function],
                              "
	",
                              [Function],
                              "
	",
                              [Function],
                              "
",
                            ],
                          },
                          "displayName": "styled.i",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-gqjmRU",
                          "target": "i",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      height=""
                      iconName="bomb"
                      id=""
                      imageFile=""
                      left=""
                      location="none"
                      noedit={false}
                      nohover={false}
                      noripple={false}
                      obj="Icon"
                      padding=""
                      selected={false}
                      sizing="normal"
                      style={Object {}}
                      testing={true}
                      theme={null}
                      tooltip=""
                      top=""
                      visible={true}
                      width=""
                      xcss={null}
                    >
                      <i
                        className="ui-icon fa fa-fw fa-bomb sc-gqjmRU iCGQxQ"
                        contentEditable={false}
                        direction="right"
                        disabled={false}
                        height=""
                        id=""
                        selected={false}
                        style={Object {}}
                        width=""
                      />
                    </StyledComponent>
                  </styled.i>
                </ThemeProvider>
              </Wrapper>
            </Icon>
            <styled.div
              justify={1}
              sizing="normal"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-chPdSV",
                      "isStatic": false,
                      "lastClassName": "eCsFtU",
                      "rules": Array [
                        "
	flex: 1;
	padding: 0 0.2em;
	text-align: ",
                        [Function],
                        ";
	",
                        [Function],
                        ";

	span {
		margin: 0 auto;
		user-select: none;
	}
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-chPdSV",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                justify={1}
                sizing="normal"
              >
                <div
                  className="sc-chPdSV eCsFtU"
                />
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </ThemeProvider>
  </Wrapper>
</ButtonText>
`;

exports[`Test making a ButtonText control invisible 1`] = `
<ButtonText
  bottom=""
  className=""
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage=""
  focus={false}
  height=""
  iconName="bomb"
  id=""
  imageFile=""
  justify={1}
  left=""
  location="none"
  noedit={false}
  nohover={false}
  noicon={false}
  noripple={false}
  obj="ButtonText"
  onClick={[MockFunction]}
  padding=""
  selected={false}
  sizing="normal"
  style={Object {}}
  testing={true}
  text=""
  theme={null}
  tooltip=""
  top=""
  visible={false}
  width=""
  xcss={null}
>
  <Wrapper
    bottom=""
    className=""
    contentEditable={false}
    controlled={true}
    direction="right"
    disabled={false}
    err={null}
    errorMessage=""
    focus={false}
    height=""
    iconName="bomb"
    id=""
    imageFile=""
    justify={1}
    left=""
    location="none"
    noedit={false}
    nohover={false}
    noicon={false}
    noripple={false}
    obj="ButtonText"
    onClick={[MockFunction]}
    onError={[Function]}
    padding=""
    reset={false}
    selected={false}
    sizing="normal"
    style={Object {}}
    testing={true}
    text=""
    theme={null}
    tooltip=""
    top=""
    visible={false}
    width=""
    xcss={null}
  >
    <ThemeProvider
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#c0c0c0",
          "color": "#000000",
          "headerBackgroundColor": "#004358",
          "headerForegroundColor": "#ffffff",
          "headerHoverColor": "#ffe11a",
          "hoverColor": "#c0c0c0",
          "inputBorderColor": "#c0c0c0",
          "itemHoverColor": "#bedb39",
          "selectedBackgroundColor": "#1f8a70",
          "selectedForegroundColor": "#ffe11a",
          "titleBarBackgroundColor": "#004358",
          "titleBarForegroundColor": "#ffe11a",
          "tooltipBackgroundColor": "#708090",
          "tooltipForegroundColor": "#ffffff",
          "transitionDelay": "0.5s",
        }
      }
    >
      <styled.div
        className="ui-button-text ripple"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        visible={false}
      >
        <StyledComponent
          className="ui-button-text ripple"
          disabled={false}
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-kgoBCf",
                "isStatic": false,
                "lastClassName": "kESvqt",
                "rules": Array [
                  "
	",
                  "
	align-items: center;
	box-sizing: border-box;
	cursor: default;
	display: flex;
	height: 100%;
	justify-content: center;
	line-height: inherit;
	outline: none;
	overflow: visible;
	position: relative;
	user-select: none;
	width: 100%;
",
                  "

	&:not(.nohover):hover {
		background-color: ",
                  [Function],
                  " ",
                  [Function],
                  ";
	}

	",
                  [Function],
                  "
	",
                  [Function],
                  "
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-kgoBCf",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          onClick={[Function]}
          style={Object {}}
          visible={false}
        >
          <div
            className="ui-button-text ripple sc-kgoBCf kESvqt"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
          >
            <Icon
              bottom=""
              className=""
              contentEditable={false}
              controlled={true}
              direction="right"
              disabled={false}
              err={null}
              errorMessage=""
              focus={false}
              height=""
              iconName="bomb"
              id=""
              imageFile=""
              left=""
              location="none"
              noedit={false}
              nohover={false}
              noripple={false}
              obj="Icon"
              padding=""
              selected={false}
              sizing="normal"
              style={Object {}}
              testing={true}
              theme={null}
              tooltip=""
              top=""
              visible={true}
              width=""
              xcss={null}
            >
              <Wrapper
                bottom=""
                className=""
                contentEditable={false}
                controlled={true}
                direction="right"
                disabled={false}
                err={null}
                errorMessage=""
                focus={false}
                height=""
                iconName="bomb"
                id=""
                imageFile=""
                left=""
                location="none"
                noedit={false}
                nohover={false}
                noripple={false}
                obj="Icon"
                onError={[Function]}
                padding=""
                reset={false}
                selected={false}
                sizing="normal"
                style={Object {}}
                testing={true}
                theme={null}
                tooltip=""
                top=""
                visible={true}
                width=""
                xcss={null}
              >
                <ThemeProvider
                  theme={
                    Object {
                      "backgroundColor": "#ffffff",
                      "borderColor": "#c0c0c0",
                      "color": "#000000",
                      "headerBackgroundColor": "#004358",
                      "headerForegroundColor": "#ffffff",
                      "headerHoverColor": "#ffe11a",
                      "hoverColor": "#c0c0c0",
                      "inputBorderColor": "#c0c0c0",
                      "itemHoverColor": "#bedb39",
                      "selectedBackgroundColor": "#1f8a70",
                      "selectedForegroundColor": "#ffe11a",
                      "titleBarBackgroundColor": "#004358",
                      "titleBarForegroundColor": "#ffe11a",
                      "tooltipBackgroundColor": "#708090",
                      "tooltipForegroundColor": "#ffffff",
                      "transitionDelay": "0.5s",
                    }
                  }
                >
                  <styled.i
                    bottom=""
                    className="ui-icon fa fa-fw fa-bomb"
                    contentEditable={false}
                    controlled={true}
                    direction="right"
                    disabled={false}
                    err={null}
                    errorMessage=""
                    focus={false}
                    height=""
                    iconName="bomb"
                    id=""
                    imageFile=""
                    left=""
                    location="none"
                    noedit={false}
                    nohover={false}
                    noripple={false}
                    obj="Icon"
                    padding=""
                    selected={false}
                    sizing="normal"
                    style={Object {}}
                    testing={true}
                    theme={null}
                    tooltip=""
                    top=""
                    visible={true}
                    width=""
                    xcss={null}
                  >
                    <StyledComponent
                      bottom=""
                      className="ui-icon fa fa-fw fa-bomb"
                      contentEditable={false}
                      controlled={true}
                      direction="right"
                      disabled={false}
                      err={null}
                      errorMessage=""
                      focus={false}
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "sc-gqjmRU",
                            "isStatic": false,
                            "lastClassName": "iCGQxQ",
                            "rules": Array [
                              "
	text-align: center;
	",
                              [Function],
                              "
	",
                              [Function],
                              "
	",
                              [Function],
                              "
	",
                              [Function],
                              "
",
                            ],
                          },
                          "displayName": "styled.i",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "sc-gqjmRU",
                          "target": "i",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      height=""
                      iconName="bomb"
                      id=""
                      imageFile=""
                      left=""
                      location="none"
                      noedit={false}
                      nohover={false}
                      noripple={false}
                      obj="Icon"
                      padding=""
                      selected={false}
                      sizing="normal"
                      style={Object {}}
                      testing={true}
                      theme={null}
                      tooltip=""
                      top=""
                      visible={true}
                      width=""
                      xcss={null}
                    >
                      <i
                        className="ui-icon fa fa-fw fa-bomb sc-gqjmRU iCGQxQ"
                        contentEditable={false}
                        direction="right"
                        disabled={false}
                        height=""
                        id=""
                        selected={false}
                        style={Object {}}
                        width=""
                      />
                    </StyledComponent>
                  </styled.i>
                </ThemeProvider>
              </Wrapper>
            </Icon>
            <styled.div
              justify={1}
              sizing="normal"
            >
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-chPdSV",
                      "isStatic": false,
                      "lastClassName": "eCsFtU",
                      "rules": Array [
                        "
	flex: 1;
	padding: 0 0.2em;
	text-align: ",
                        [Function],
                        ";
	",
                        [Function],
                        ";

	span {
		margin: 0 auto;
		user-select: none;
	}
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-chPdSV",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                justify={1}
                sizing="normal"
              >
                <div
                  className="sc-chPdSV eCsFtU"
                />
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </ThemeProvider>
  </Wrapper>
</ButtonText>
`;

exports[`Test retrieval of ButtonText props object 1`] = `
Object {
  "bottom": "",
  "children": null,
  "className": "",
  "contentEditable": false,
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": "",
  "focus": false,
  "height": "",
  "iconName": "bomb",
  "id": "",
  "imageFile": "",
  "justify": 1,
  "left": "",
  "location": "none",
  "noedit": false,
  "nohover": false,
  "noicon": false,
  "noripple": false,
  "obj": "ButtonText",
  "onClick": [Function],
  "padding": "",
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "testing": true,
  "text": "",
  "theme": null,
  "tooltip": "",
  "top": "",
  "visible": true,
  "width": "",
  "xcss": null,
}
`;
